features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/net/RumRequestFactory.kt

```diff
@@ -73,40 +86,90 @@ internal class RumRequestFactory(
 
     private fun buildHeaders(
         requestId: String,
-        clientToken: String,
-        source: String,
-        sdkVersion: String
+        idempotencyKey: String?,
+        context: DatadogContext
     ): Map<String, String> {
-        return mapOf(
-            RequestFactory.HEADER_API_KEY to clientToken,
-            RequestFactory.HEADER_EVP_ORIGIN to source,
-            RequestFactory.HEADER_EVP_ORIGIN_VERSION to sdkVersion,
+        val headers = mutableMapOf(
+            RequestFactory.HEADER_API_KEY to context.clientToken,
+            RequestFactory.HEADER_EVP_ORIGIN to context.source,
+            RequestFactory.HEADER_EVP_ORIGIN_VERSION to context.sdkVersion,
             RequestFactory.HEADER_REQUEST_ID to requestId
         )
+        if (idempotencyKey != null) {
+            headers[RequestFactory.DD_IDEMPOTENCY_KEY] = idempotencyKey
+        }
+        return headers
     }
 
     private fun buildTags(
         serviceName: String,
         version: String,
         sdkVersion: String,
         env: String,
-        variant: String
+        variant: String,
+        executionContext: RequestExecutionContext
     ): String {
         val elements = mutableListOf(
             "${RumAttributes.SERVICE_NAME}:$serviceName",
             "${RumAttributes.APPLICATION_VERSION}:$version",
             "${RumAttributes.SDK_VERSION}:$sdkVersion",
-            "${RumAttributes.ENV}:$env"
+            "${RumAttributes.ENV}:$env",
+            "${RETRY_COUNT_KEY}:${executionContext.attempts}"
         )
 
         if (variant.isNotEmpty()) {
             elements.add("${RumAttributes.VARIANT}:$variant")
         }
+        if (executionContext.previousResponseCode != null) {
+            elements.add("${LAST_FAILURE_STATUS_KEY}:${executionContext.previousResponseCode}")
+        }
 
         return elements.joinToString(",")
     }
 
+    @Suppress("TooGenericExceptionCaught")
```

Comment:
is this `@Supress` necessary? it seems you catch all the possible exceptions

Reply:
good questions...it was before I added this in the `detekt.xml` let me see it now.

Reply:
yes it is still complaining and due to the `NullPointerException` catch at the end of the block. I am not sure why though as this is specifically thrown by the code under the hood.

Reply:
It's a default set of detekt to include NPE in this check, because for Kotlin code, NPE is not very informative, but since we are calling a Java function, there is nothing we can do about, so it's correct to have `@Suppress` here I think.

https://detekt.dev/docs/rules/exceptions/#toogenericexceptioncaught